### DISCUSSION TITLE
Getting Definition name for schema from DereferencedRequest

### DISCUSSION THREAD

**Discussion by nexon** [mattpolzin/OpenAPIKit#289] [Q&A]
First of all, Great Library. Is quite easy to use. So congrats for the great work.

I have a question regarding how to obtain the Definition name for a Schema when looping through the **content** (DereferencedContent.Map).

I can obtain obviously the schema with all the info except the name for it. that's the only thing that i'm missing. Do we have a direct way of obtaining it?.

Thanks!

**Comment by mattpolzin** [comment]
When you say Definition name are you referring to the name a schema was stored under in the Components Object prior to dereferencing?

**Reply by nexon** [reply]
Exactly. Right now i don't see an easy way of getting it.

**Reply by mattpolzin** [reply]
This is not something currently kept track of when dereferencing using the builtin `dereferenced(in:)` functions. It's one way in which I have recently been thinking through improving upon the dereferencing functionality, though. I think I want to store the name something was originally referenced at within the Components object as a vendor extension on the dereferenced object. That also means needing to introduce vendor extension support to the JSON Schema type, which is a longstanding TODO.

I may try to get a solution for this in place sooner than later since it's been on my radar anyway and now you've brought it up as well.

**Reply by nexon** [reply]
Well, if I can help in some way, just let me know. Happy to help

**Reply by mattpolzin** [reply]
Just to follow up with a long overdue update since I forgot to comment here at the time, this has since been implemented as described in the README [here](https://github.com/mattpolzin/OpenAPIKit/blob/main/README.md:419). I named the extension `x-component-name`.
